Sources/IfTag.swift

```diff
@@ -100,21 +107,72 @@ final class IfExpressionParser {
   let tokens: [IfToken]
   var position: Int = 0
 
-  init(components: [String], tokenParser: TokenParser) throws {
-    self.tokens = try components.map { component in
-      if let op = findOperator(name: component) {
-        switch op {
-        case .infix(let name, let bindingPower, let cls):
-          return .infix(name: name, bindingPower: bindingPower, op: cls)
-        case .prefix(let name, let bindingPower, let cls):
-          return .prefix(name: name, bindingPower: bindingPower, op: cls)
+  private init(tokens: [IfToken]) {
+    self.tokens = tokens
+  }
+  
+  static func parser(components: [String], tokenParser: TokenParser) throws -> IfExpressionParser {
+    try IfExpressionParser.validateBracketsBalance(components: components)
+    return try IfExpressionParser(components: ArraySlice(components), tokenParser: tokenParser)
+  }
+
+  static func validateBracketsBalance(components: [String]) throws {
+    guard try components.reduce(0, {
+      var bracketsBalance = $0
+      if $1 == "(" { bracketsBalance += 1 }
+      else if $1 == ")" { bracketsBalance -= 1 }
+      if bracketsBalance < 0 { throw TemplateSyntaxError("unbalanced brackets: too many closing brackets") }
```

Comment:
Sorry to be nitpicking about code style, but if we soon add SwiftLint to harmonize the code style of all contributors it will raise a violation here probably: even for one-liners like that it'll suggest to go to the next line after opening brace and before closing brace. Besides just being strict with style (which I agree can be annoying sometimes) this has the advantage of allowing to debug step by step, entering inside the scope of needed during debug.